app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt

```diff
@@ -101,6 +107,43 @@ class MessageActivity : AppCompatActivity() {
             }
         }
 
+        save.setOnClickListener{
+            try {
+                val inputManager: InputMethodManager = this.getSystemService(Context.INPUT_METHOD_SERVICE)
+                        as InputMethodManager
+                inputManager.hideSoftInputFromWindow(content.windowToken, InputMethodManager.SHOW_FORCED)
+
+                val sdf = SimpleDateFormat("dd-MMM-yyyy-hh_mm_aaa")
+                val fileName = "badge-magic-" + sdf.format(Calendar.getInstance().time).toString()
+                val file = File(applicationContext.filesDir, fileName)
+                val qFile = QueueFile.Builder(file).build()
+
+                if(content.text.isNotBlank())
+                    qFile.add(("Content-Text - "+content.text.toString()).toByteArray())
+                else if(drawableRecyclerAdapter.selectedPosition >= 0)
+                    qFile.add(("Content-Drawable - "+drawableRecyclerAdapter.selectedPosition).toByteArray())
+
+                qFile.add(("Speed - "+speed.selectedItem.toString()).toByteArray())
+                qFile.add(("Mode - "+mode.selectedItem.toString()).toByteArray())
+
+                if(flash.isChecked)
+                    qFile.add("Type - Flash".toByteArray())
+                else if(marquee.isChecked)
+                    qFile.add("Type - Marquee".toByteArray())
+                else
+                    qFile.add("Type - None".toByteArray())
+
+                val sb = StringBuilder()
+                sb.append("Records saved successfully with file name ")
+                sb.append(fileName)
+
+                Toast.makeText(this,sb.toString(),Toast.LENGTH_SHORT).show()
```

Comment:
Use POJO and JSON for saving items, and don't create a file for each item

Reply:
@iamareebjamal Will it be okay if I and @yatri1609 work on the same issue and then you can decide which PR can be merged.